seacrowd/sea_datasets/vivqa/vivqa.py

```diff
@@ -0,0 +1,132 @@
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_imqa",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features({"img_id": datasets.Value("string"),
+                                          "question": datasets.Value("string"),
+                                          "answer": datasets.Value("string"),
+                                          "type": datasets.Value("string")})
+        elif self.config.schema == "seacrowd_imqa":
+            features = schemas.imqa_features
+            features['meta'] = {"coco_img_id": datasets.Value("string")}
+        else:
+            raise ValueError(f"No schema matched for {self.config.schema}")
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS["viviq"]
+        data_dir = dl_manager.download_and_extract(urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": data_dir["train"],
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": data_dir["test"],
+                    "split": "test",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        raw_examples = pd.read_csv(filepath)
+
+        for eid, exam in raw_examples.iterrows():
+            assert len(exam) == 5
+            exam_id, exam_quest, exam_answer, exam_img_id, exam_type = exam
+
+            if self.config.schema == "source":
+                yield eid, {"img_id": str(exam_img_id),
+                            "question": exam_quest,
+                            "answer": exam_answer,
+                            "type": exam_type}
+            elif self.config.schema == "seacrowd_imqa":
+                example = {
+                    "id": str(eid),
+                    "question_id": str(exam_id),
+                    "document_id": str(eid),
+                    "questions": [exam_quest],
+                    "type": exam_type,
+                    "choices": None,
+                    "context": str(exam_img_id),
+                    "answer": [exam_answer],
+                    "image_paths": [exam_img_id],
```

Comment:
Use full path to the image not only the ids.

Reply:
Hi, @holylovenia, the original data only contains a coco-dataset image id, shall we still load the image full path for this data, as the coco api (cannot install with simple pip or conda) is needed to retrive the image path .

Reply:
Hmm I see, @Gyyz. Could the images be taken from here?

```
"coco2017_train_images": "http://images.cocodataset.org/zips/train2017.zip",
"coco2014_train_images": "http://images.cocodataset.org/zips/train2014.zip",
"coco2014_val_images": "http://images.cocodataset.org/zips/val2014.zip",
```